app/api/groups/delete_group.go

```diff
@@ -0,0 +1,78 @@
+package groups
+
+import (
+	"net/http"
+
+	"github.com/go-chi/render"
+
+	"github.com/France-ioi/AlgoreaBackend/app/database"
+	"github.com/France-ioi/AlgoreaBackend/app/service"
+)
+
+// swagger:operation DELETE /groups/{group_id} groups groupDelete
+// ---
+// summary: Delete a group
+// description: >
+//   Removes a group, its parent-child relations, and recursively deletes each
+//   new orphaned group.
```

Comment:
The spec was "The group must be empty (no active subgroups of any type)" so I don't get why you would delete any orphaned group. 🤔

Reply:
Fixed. I was confused by the phrase 'with all cascading as done in groupRemoveChild with delete_orphans'.